src/components/creeps/harvester.ts

```diff
@@ -21,9 +21,11 @@ export default class Harvester extends CreepAction implements IHarvester, ICreep
 
   public setCreep(creep: Creep) {
     super.setCreep(creep);
-
-    this.targetSource = Game.getObjectById<Source>(this.creep.memory.target_source_id);
-    this.targetEnergyDropOff = Game.getObjectById<Spawn | Structure>(this.creep.memory.target_energy_dropoff_id);
+    let source = Game.getObjectById<Source>(this.creep.memory.target_source_id);
+    let dropOff = Game.getObjectById<Spawn | Structure>(this.creep.memory.target_energy_dropoff_id);
+    if (source === null || dropOff === null) { return false; };
```

Comment:
* `return false;` should be on its own line, and remove the trailing semicolon outside of the curly brace, e.g.

```js
if (source === null || dropOff === null) {
  return false;
}
```